### PR TITLE
[feat] Makefile release taging refactor

### PR DESCRIPTION
**Pull request by phoinixgrr**
#### Summary

We are changing the logic that the APP_VERSION variable is being generated.
Specifically:

- If we run under a protected_branch, we use `protected_branch_name-sha` as APP_VERSION
- If we run under a semver tag, we also push `latest` tag

Signed-off-by: Akis Maziotis <akis.maziotis@mattermost.com>

#### Ticket Link

Ticket: https://mattermost.atlassian.net/browse/DOPS-908

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -146,9 +146,14 @@ docker-build: ## to build the docker image
 .PHONY: docker-push
 docker-push: ## to push the docker image
 	@$(INFO) Pushing to registry...
-	$(AT)$(DOCKER) tag ${APP_NAME}:${APP_VERSION} $(DOCKER_REGISTRY)/${DOCKER_REGISTRY_REPO}/${APP_NAME}:${APP_VERSION} || ${FAIL}
-	$(AT)$(DOCKER) push $(DOCKER_REGISTRY)/${DOCKER_REGISTRY_REPO}/${APP_NAME}:${APP_VERSION} || ${FAIL}
-	@$(OK) Pushing to registry $(DOCKER_REGISTRY)/${DOCKER_REGISTRY_REPO}/${APP_NAME}:${APP_VERSION}
+	$(AT)$(DOCKER) tag ${APP_NAME}:${APP_VERSION} $(DOCKER_REGISTRY)/${DOCKER_REGISTRY_REPO}:${APP_VERSION} || ${FAIL}
+	$(AT)$(DOCKER) push $(DOCKER_REGISTRY)/${DOCKER_REGISTRY_REPO}:${APP_VERSION} || ${FAIL}
+# if we are on a tag, also push latest
+ifneq ($(shell echo $(APP_VERSION) | egrep '^v([0-9]+\.){0,2}(\*|[0-9]+)'),)
```
Comment (pfltdv) [inline Makefile:152]: What if we are at older tag. For instance assume that we have a 6.6.0 release yesterday and it was pushed as latest.

Today what if we decided to publish 6.5.1 patch release. Latest tag should still point to 6.6.0.

Comment (phoinixgrr) [reply Makefile:152]: That's a good point. I've assumed that release always iterates on the previous release. 
Is that a possibility?

Comment (phoinixgrr) [reply Makefile:152]: I guess it is , if we push a patch on a previous minor version. 
Will refactor the code

Comment (phoinixgrr) [reply Makefile:152]: Rebased and pushed changes

Comment (pfltdv) [reply Makefile:152]: :heart: Nice one!

**Review by pfltdv — APPROVED** [review]
:100: Great work.

**Pull request merged**